sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashJoin.scala

```diff
@@ -316,6 +318,387 @@ trait HashJoin extends BaseJoinExec {
       resultProj(r)
     }
   }
+
```

Comment:
can you use PR comments to highlight the real changes? Seems most of the diff is just moving code around.

Reply:
@cloud-fan - sorry about that. yes mostly of them is moving code around without change. Highlighted change with comments, thanks.